src/PostgREST/Types.hs

```diff
@@ -0,0 +1,110 @@
+module PostgREST.Types where
+import Data.Text
+import Data.Tree
+import qualified Data.ByteString.Char8 as BS
+import Data.Aeson
+
+data DbStructure = DbStructure {
+  tables :: [Table]
+, columns :: [Column]
+, relations :: [Relation]
+, primaryKeys :: [PrimaryKey]
+--, tablesAcl :: [(Text, Text, Text)]
+}
+
+
+data Table = Table {
+  tableSchema :: Text
+, tableName :: Text
+, tableInsertable :: Bool
+, tableAcl :: [Text]
+} deriving (Show)
+
+data ForeignKey = ForeignKey {
+  fkTable::Text, fkCol::Text
+} deriving (Show)
+
+
+data Column = Column {
+  colSchema :: Text
+, colTable :: Text
+, colName :: Text
+, colPosition :: Int
+, colNullable :: Bool
+, colType :: Text
+, colUpdatable :: Bool
+, colMaxLen :: Maybe Int
+, colPrecision :: Maybe Int
+, colDefault :: Maybe Text
+, colEnum :: [Text]
+, colFK :: Maybe ForeignKey
+} | Star {colSchema :: Text, colTable :: Text } deriving (Show)
+
+data PrimaryKey = PrimaryKey {
+  pkSchema::Text, pkTable::Text, pkName::Text
+}
+
+data OrderTerm = OrderTerm {
+  otTerm :: Text
+, otDirection :: BS.ByteString
+, otNullOrder :: Maybe BS.ByteString
+} deriving (Show, Eq)
+
+
+data Relation = Relation {
+  relSchema  :: Text
+, relTable   :: Text
+, relColumn  :: Text
+, relFTable  :: Text
+, relFColumn :: Text
+, relType    :: Text
```

Comment:
What if

```haskell
relType :: RelationType
```
where
```haskell
data RelationType = RelChild | RelParent | RelMany
```

It looks like it is functioning as an enum. If we get its possible values out of strings it would make better use of the type system.